components/arm/trossen/trossen.go

```diff
@@ -231,17 +246,113 @@ func (a *Arm) JointPositions(ctx context.Context, extra map[string]interface{})
 		return &pb.JointPositions{}, err
 	}
 
-	positions := make([]float64, 0, len(a.JointOrder()))
-	for _, jointName := range a.JointOrder() {
+	numJoints := len(a.jointOrder())
+	positions := make([]float64, 0, numJoints)
+	for _, jointName := range a.jointOrder() {
 		positions = append(positions, servoPosToValues(angleMap[jointName]))
 	}
 
 	return &pb.JointPositions{Values: positions}, nil
 }
 
-// Stop is unimplemented for trossen.
+// OpenGripper opens the gripper.
+func (a *Arm) OpenGripper(ctx context.Context) error {
+	ctx, done := a.opMgr.New(ctx)
+	defer done()
+	a.moveLock.Lock()
+	defer a.moveLock.Unlock()
+
+	pos, err := a.Joints["Gripper"][0].PresentPosition()
+	if err != nil {
+		return errors.Wrap(err, "position retrieval failed when opening gripper")
+	}
+	if pos >= 2800 {
+		a.logger.Debug("gripper already open, returning")
+		return nil
+	}
+
+	err = a.Joints["Gripper"][0].SetGoalPWM(150)
+	if err != nil {
+		return err
+	}
+	a.logger.Debug("gripper pwm set to 150")
+
+	// We don't want to over-open
+	atPos := false
+	for !atPos {
+		var pos int
+		pos, err = a.Joints["Gripper"][0].PresentPosition()
+		if err != nil {
+			return errors.Wrap(err, "position retrieval failed when opening gripper")
+		}
+		if pos < 2800 {
+			if !utils.SelectContextOrWait(ctx, 50*time.Millisecond) {
+				return ctx.Err()
+			}
+		} else {
+			atPos = true
+			a.logger.Debug("reached open gripper position")
+		}
+	}
+	err = a.Joints["Gripper"][0].SetGoalPWM(0)
+	if err != nil {
+		a.logger.Debug("gripper pwm set to 0")
```

Comment:
should return an err? or log a warn? logging at debug level for an err event seems too lax.

Reply:
ah yes this was a mistake will fix